content/doc/book/pipeline/jenkinsfile.adoc

```diff
@@ -4,145 +4,469 @@ layout: section
 :notitle:
 :description:
 :author:
-:email: jenkinsci-users@googlegroups.com
+:email: jenkinsci-docs@googlegroups.com
 :sectanchors:
-:toc: left
+:toc:
+:hide-uri-scheme:
 
 = The Jenkinsfile
 
-== Audience and Purpose
+This section builds on the information covered in <<getting-started#, Getting Started>>,
+and introduces more useful steps, common patterns and demonstrates some
+non-trivial `Jenkinsfile` examples.
+
+Creating a `Jenkinsfile`, which is checked into source control
+footnoteref:[scm, https://en.wikipedia.org/wiki/Source_control_management],
+provides a number of immediate benefits:
+
+* Code review/iteration on the Pipeline
+* Audit trail for the Pipeline
+* Single source of truth
+  footnote:[https://en.wikipedia.org/wiki/Single_Source_of_Truth]
+  for the Pipeline, which can be viewed and edited by multiple members of the project.
+
+
+While the syntax for defining a Pipeline, either in the web UI or with a
+`Jenkinsfile`, is the same, it's generally considered best practice to define
+the Pipeline in a `Jenkinsfile` and check that in to source control.
 
-This document is intended for Jenkins users who want to leverage the power of
-pipeline functionality. Extending the reach of what was learned from a "Hello
-World" example in link:/doc/pipeline/[Getting Started with Pipeline], this
-document explains how to use a `Jenkinsfile` to perform a simple checkout and
-build for the contents of a repository.
 
 == Creating a Jenkinsfile
 
-A `Jenkinsfile` is a container for your pipeline (or other) script, which details
-what specific steps are needed to perform a job for which you want to use
-Jenkins. You create a `Jenkinsfile` with your preferred Groovy editor, or through
-the configuration page on the web interface of your Jenkins instance.
+As discussed in the <<getting-started#defining-a-pipeline-in-scm, Getting Started>>
+section, a `Jenkinsfile` is a text file that contains the definition of a
+Jenkins Pipeline and is checked into source control. Consider the following
+Pipeline which implements a basic, three-stage, continuous delivery pipeline.
 
-Using a Groovy editor to code a `Jenkinsfile` gives you more flexibility for
-building complex single or multibranch pipelines, but whether you use an editor
-or the Jenkins interface does not matter if what you want to do is get familiar
-with basic `Jenkinsfile` content.
+[pipeline]
+----
+// Script //
+node { // <1>
+    stage('Build') { // <2>
+        /* .. snip .. */
+    }
+    stage('Test') {
+        /* .. snip .. */
+    }
+    stage('Deploy') {
+        /* .. snip .. */
+    }
+}
+----
+<1> `node` allocates an executor and workspace on the Jenkins cluster.
```

Comment:
I'm not sure, I have my doubts about the phrase "Jenkins cluster" despite how often I use it when speaking.

The intent is to capture the concept of "an executor on the master or any given agent".  Previous iterations of this documentation have used the phrase "agent pool" which I discarded since that implicitly excludes the master.